### DISCUSSION TITLE
"Use same as above" checkbox to mirror previous fields

### DISCUSSION THREAD

**Discussion by sadarak** [react-hook-form/react-hook-form#2777] [Q&A] [answered]
Hi! 

I am wondering how I should implement the pattern that you often get in a form where you can select a checkbox to make it mirror some information you have already typed previously in the form. This is often used for e.g. shipping and billing address. 

Example:
[Name input]
[Email input]

[Checkbox "Use same as above?"] (if checked, mirrors the previous name and email in the fields below and disables them)
[Name input]
[Email input]

Suggestions on how this should be implemented?

**Comment by bluebill1049** [comment, marked as answer]
https://react-hook-form.com/api#watch

**Reply by sadarak** [reply]
Thank you, now how would you mirror the value? The fields already has a default value coming from the state, since the form is a multi step proccess(user can go back and forth). I tried setting defaultValue, but I guess that doesn't have any effect when the fields get the default value from defaultValues: {} useForm.

**Reply by bluebill1049** [reply]
provide a simple CSB

**Reply by sadarak** [reply]
CSB? I guess what I am asking is how do I mirror the value, I can use setValue when the user clicks the checkbox, but then if the user make changes to the first fields, the latter fields won't get updated with the same values.

**Reply by bluebill1049** [reply]
codesandbox.

**Reply by sadarak** [reply]
https://1680p.codesandbox.io/

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/cool-dew-1pvsw?file=/src/Contact.tsx

**Reply by sadarak** [reply]
Thank you, useEffect was my missing piece.

**Reply by bluebill1049** [reply]
np
